Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp

```diff
@@ -414,14 +414,19 @@ void WebAutomationSessionProxy::evaluateJavaScriptFunction(WebCore::PageIdentifi
         toJSValue(context, function),
         toJSArray(context, arguments, toJSValue, &exception),
         JSValueMakeBoolean(context, expectsImplicitCallbackArgument),
+        JSValueMakeBoolean(context, forceUserGesture),
         JSValueMakeNumber(context, frameID.object().toUInt64()),
         JSValueMakeNumber(context, frameID.processIdentifier().toUInt64()),
         JSValueMakeNumber(context, callbackID),
         JSObjectMakeFunctionWithCallback(context, nullptr, evaluateJavaScriptCallback),
         JSValueMakeNumber(context, callbackTimeout.value_or(-1))
     };
 
-    callPropertyFunction(context, scriptObject, "evaluateJavaScriptFunction"_s, std::size(functionArguments), functionArguments, &exception);
+    if (forceUserGesture) {
+        WebCore::UserGestureIndicator gestureIndicator { WebCore::ProcessingUserGesture, frame->coreLocalFrame()->document() };
+        callPropertyFunction(context, scriptObject, "evaluateJavaScriptFunction"_s, std::size(functionArguments), functionArguments, &exception);
```

Comment:
You can use `emplace()` to avoid duping `callPropertyFunction`

Reply:
Hmm... sorry I don't follow. Use `emplace()` how?

Reply:
```
std::optional<WebCore::UserGestureIndicator> gestureIndicator;
if (forceUserGesture)
    gestureIndicator.emplace({ WebCore::ProcessingUserGesture, frame->coreLocalFrame()->document() });
callPropertyFunction(context, scriptObject, "evaluateJavaScriptFunction"_s, std::size(functionArguments), functionArguments, &exception);

Reply:
Makes sense, I'll change to this. Thanks!